catalog/spatial/registry/registry-source-configuration-handler/src/test/java/org/codice/ddf/registry/source/configuration/SourceConfigurationHandlerTest.java

```diff
@@ -0,0 +1,842 @@
+        sch.setSourceActivationPriorityOrder(order1);
+        verify(adminService, times(1)).getRegistryMetacards();
+        sch.setSourceActivationPriorityOrder(order2);
+        verify(adminService, times(2)).getRegistryMetacards();
+    }
+
+    @Test
+    public void testEmptyMetadata() throws Exception {
+        mcard.setMetadata("");
+        setupSerialExecutor();
+        sch.handleEvent(createEvent);
+        verify(configAdmin, never()).listConfigurations(anyString());
+    }
+
+    @Test
+    public void testDestroy() throws Exception {
+        when(executorService.awaitTermination(anyLong(), any(TimeUnit.class))).thenReturn(true);
+        sch.destroy();
+        verify(executorService, times(1)).awaitTermination(anyLong(), any(TimeUnit.class));
+        verify(executorService, times(0)).shutdownNow();
+    }
+
+    @Test
+    public void testDestroyTerminateTasks() throws Exception {
+        when(executorService.awaitTermination(anyLong(), any(TimeUnit.class))).thenReturn(false);
+        sch.destroy();
+        verify(executorService, times(2)).awaitTermination(anyLong(), any(TimeUnit.class));
+        verify(executorService, times(1)).shutdownNow();
+    }
+
+    @Test
+    public void testDestroyInterupt() throws Exception {
+        when(executorService.awaitTermination(anyLong(),
+                any(TimeUnit.class))).thenThrow(new InterruptedException("interrupt"));
+        sch.destroy();
+        verify(executorService, times(1)).awaitTermination(anyLong(), any(TimeUnit.class));
+        verify(executorService, times(1)).shutdownNow();
+    }
+
+    private String getMetadata(String path) throws Exception {
+        InputStream inputStream = getClass().getResourceAsStream(path);
+        BufferedReader buffer = new BufferedReader(new InputStreamReader(inputStream));
+        return buffer.lines()
+                .collect(Collectors.joining("\n"));
+    }
+
+    //taken from org.apache.felix.webconsole.internal.servlet.ConfigurationMetatypeSupport
+    private static class AttributeDefinitionImpl implements AttributeDefinition {
+
+        private final String id;
+
+        private final String name;
+
+        private final String description;
+
+        private final int type;
+
+        private final String[] defaultValues;
+
+        private final int cardinality;
+
+        private final String[] optionLabels;
+
+        private final String[] optionValues;
+
+        AttributeDefinitionImpl(final String id, final String name, final String description,
+                final String defaultValue) {
+            this(id, name, description, STRING, new String[] {defaultValue}, 0, null, null);
+        }
+
+        AttributeDefinitionImpl(final String id, final String name, final String description,
+                final int type, final String[] defaultValues, final int cardinality,
+                final String[] optionLabels, final String[] optionValues) {
+            this.id = id;
```

Comment:
This is a copy past of the real private class that apache felix uses. I just lifted it for testing.